packages/core/src/ui/Base/CustomLink.tsx

```diff
@@ -1,22 +1,47 @@
 import Link from "next/link.js";
+import { Tooltip } from "../Tooltip";
+
+type Config = {
+  showLinkPreviews: boolean;
+};
 
 interface Props {
   href?: string;
+  data: any;
+  usehook: any;
+  config: Config;
   [x: string]: unknown;
 }
```

Comment:
`CustomLink` component doesn't need the whole `siteConfig` object. It only needs to know whether it should wrap a link in a `Toolip` or not. E.g.:
```suggestion
interface Props {
  href?: string;
  data: any;
  usehook: any;
  preview: boolean;
  [x: string]: unknown;
}
```

Reply:
Done :)